spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/FillNullsRuleInterpreter.scala

```diff
@@ -41,22 +44,33 @@ case class FillNullsRuleInterpreter(rule: FillNullsConformanceRule) extends Rule
       rule.outputColumn
     )
 
+    val dataType = SchemaUtils.getFieldType(rule.inputColumn, df.schema).get
+    val default: Column = simpleLiteralCast(rule.value, dataType) match {
```

Comment:
Sorry, I might be out of context. But why not
```scala
val default: Column = lit(rule.value).cast(dataType)
```

Reply:
We do not support all DataTypes, officially. But they might pass through standardization. So from this point of view, it is to be on the safe side.

Also, there were ideas of passing in some extra metadata (to almost all rules) and this was what I was left with.